main-cookbooks/teracy-dev/recipes/apache.rb

```diff
@@ -12,10 +12,41 @@
     node.default['apache']['listen_ports'] = [node['teracy-dev']['apache']['listen_port']]
     node.default['apache']['user'] = 'vagrant'
     node.default['apache']['group'] = 'vagrant'
+    node.default['apache']['version'] = node['teracy-dev']['apache']['version']
 
     apache_installed = Mixlib::ShellOut.new('which apache2').run_command.stdout
+    apache_version =  Mixlib::ShellOut.new('apache2 -v | grep -o "[0-9]\.[0-9]"').run_command.stdout
+    if apache_installed == '' or apache_version != node['teracy-dev']['apache']['version']
+        bash 'clean up previous install apapche' do
+            code <<-EOF
+                apt-get purge apache2 -y -f
+                apt-get purge apache2* -y -f
+                apt-get purge apache2 apache2-utils apache2.2-bin apache2-common -y -f
+                apt-get install -y python-software-properties
+                apt-get autoremove
+                rm -rf /etc/apache2 && rm -rf /usr/lib/apache2
+            EOF
+            user 'root'
+        end
+        if node['teracy-dev']['apache']['version'] == '2.2'
+            bash 'remove apache 2.4 ppa' do
+                code <<-EOF
+                    add-apt-repository -y --remove ppa:ondrej/apache2
+                    apt-get update
+                EOF
+                user 'root'
+            end
+        else
+            bash 'add apache 2.4 ppa' do
+                code <<-EOF
+                    add-apt-repository -y ppa:ondrej/apache2
```

Comment:
Not tested yet but that possible. Not sure about any side effect compare to command line. But i feel more trust the bash command.